src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs

```diff
@@ -468,4 +470,49 @@ public void should_not_throw_exception_on_valid_field_name_using_field_builder(s
 
         type.Fields.Last().Name.ShouldBe(fieldName);
     }
+
+    [Fact]
+    public void create_fieldbuilder_with_inferred_graph_type()
+    {
+        var type = new ComplexType<TestObject>();
+
+        type.Field<int>("field1").Resolve(_ => 3);
+        type.Fields.Find("field1").ShouldNotBeNull().Type.ShouldBe(typeof(NonNullGraphType<GraphQLClrOutputTypeReference<int>>));
+
+        type.Field<int>("field2", false).Resolve(_ => 3);
+        type.Fields.Find("field2").ShouldNotBeNull().Type.ShouldBe(typeof(NonNullGraphType<GraphQLClrOutputTypeReference<int>>));
+
+        type.Field<int>("field3", true).Resolve(_ => 3);
+        type.Fields.Find("field3").ShouldNotBeNull().Type.ShouldBe(typeof(GraphQLClrOutputTypeReference<int>));
+
+        Should.Throw<ArgumentException>(() => type.Field<int?>("field4"));
```

Comment:
Please add `.Message.ShouldBe(...)` to make this test more readable.

Reply:
done